service/history/execution/mutable_state_builder.go

```diff
@@ -2174,12 +2166,6 @@ func (e *mutableStateBuilder) AddActivityTaskScheduledEvent(
 		activityStartedScope.IncCounter(metrics.CadenceRequests)
 		return event, ai, nil, true, true, nil
 	}
-	// TODO merge active & passive task generation
```

Comment:
This will change the behavior of activity local dispatch.

Before, we try dispatch activity task and if it succeeds, we skip the transfer task creation.
After, we always create the transfer task.

Reply:
Yes, updated. Good catch.